core-bundle/src/ServiceAnnotation/Hook.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\ServiceAnnotation;
+
+use Doctrine\Common\Annotations\Annotation\Attribute;
+use Doctrine\Common\Annotations\Annotation\Attributes;
+use Doctrine\Common\Annotations\Annotation\Target;
+use Terminal42\ServiceAnnotationBundle\Annotation\ServiceTag;
+
+/**
+ * Annotation that can be used to register a Contao hook.
+ *
+ * @Annotation
+ * @Target({"METHOD"})
+ * @Attributes({
+ *     @Attribute("hook", type="string", required=true),
+ *     @Attribute("priority", type="int"),
+ * })
+ */
+final class Hook extends ServiceTag
```

Comment:
Final classes cannot be tested, therefore we should not make classes final.

Reply:
They can be tested just fine, they just cannot be mocked, which we don't need to 😉

Reply:
Why do they have to be final at all?

Reply:
They're annotations. It never makes sense to merge them, the class annotations that define the annotation would not be parsed from parent classes.

Reply:
Ok